src/Test/Utilities/Desktop/CommonDiagnosticAnalyzers.cs

```diff
@@ -91,67 +81,69 @@ public static string GetExpectedErrorLogIssuesText(Compilation compilation)
+            GetExpectedPropertiesMapText() + @"
+          }
+        },
+        {
+          ""ruleId"": """ + Descriptor2.Id + @""",
+          ""locations"": [
+          ],
+          ""shortMessage"": """ + Descriptor2.MessageFormat + @""",
+          ""fullMessage"": """ + Descriptor2.Description + @""",
+          ""properties"": {
+            ""severity"": """ + Descriptor2.DefaultSeverity + @""",
+            ""defaultSeverity"": """ + Descriptor2.DefaultSeverity + @""",
+            ""title"": """ + Descriptor2.Title + @""",
+            ""category"": """ + Descriptor2.Category + @""",
+            ""helpLink"": """ + Descriptor2.HelpLinkUri + @""",
+            ""isEnabledByDefault"": """ + Descriptor2.IsEnabledByDefault + @""",
+            ""isSuppressedInSource"": ""False"",
+            ""customTags"": """ + Descriptor2.CustomTags.Join(";") + @"""" +
+            GetExpectedPropertiesMapText() + @"
+          }
         }
-      ],
-      ""shortMessage"": """ + Descriptor1.MessageFormat + @""",
-      ""fullMessage"": """ + Descriptor1.Description + @""",
-      ""properties"": {
-        ""severity"": """ + Descriptor1.DefaultSeverity + @""",
-        ""warningLevel"": ""1"",
-        ""defaultSeverity"": """ + Descriptor1.DefaultSeverity + @""",
-        ""title"": """ + Descriptor1.Title + @""",
-        ""category"": """ + Descriptor1.Category + @""",
-        ""helpLink"": """ + Descriptor1.HelpLinkUri + @""",
-        ""isEnabledByDefault"": """ + Descriptor1.IsEnabledByDefault + @""",
-        ""isSuppressedInSource"": ""False"",
-        ""customTags"": """ + Descriptor1.CustomTags.Join(";") + @"""" +
-        GetExpectedPropertiesMapText() + @"
-      }
-    },
-    {
-      ""ruleId"": """ + Descriptor2.Id + @""",
-      ""locations"": [
-      ],
-      ""shortMessage"": """ + Descriptor2.MessageFormat + @""",
-      ""fullMessage"": """ + Descriptor2.Description + @""",
-      ""properties"": {
-        ""severity"": """ + Descriptor2.DefaultSeverity + @""",
-        ""defaultSeverity"": """ + Descriptor2.DefaultSeverity + @""",
-        ""title"": """ + Descriptor2.Title + @""",
-        ""category"": """ + Descriptor2.Category + @""",
-        ""helpLink"": """ + Descriptor2.HelpLinkUri + @""",
-        ""isEnabledByDefault"": """ + Descriptor2.IsEnabledByDefault + @""",
-        ""isSuppressedInSource"": ""False"",
-        ""customTags"": """ + Descriptor2.CustomTags.Join(";") + @"""" +
-        GetExpectedPropertiesMapText() + @"
-      }
+      ]
     }
   ]
 }";
             }
 
-            public static string EscapeDirectorySeparatorChar(string input)
+            public static string GetEscapedUriForPath(string path)
             {
-                return input.Replace(Path.DirectorySeparatorChar.ToString(), @"\" + Path.DirectorySeparatorChar);
+                return new Uri(path, UriKind.RelativeOrAbsolute).ToString().Replace("/", "\\/");
```

Comment:
Do you have to escape the forward slash. The JSON spec ECMA 404 says:
```
A string is a sequence of Unicode code points wrapped with quotation marks (U+0022).
All characters may be placed within the quotation marks except for the characters that must be escaped:
quotation mark (U+0022), reverse solidus (U+005C), and the control characters U+0000 to U+001F.
There are two-character escape sequence representations of some characters.  

    \" represents the quotation mark character (U+0022).
    \\ represents the reverse solidus character (U+005C).
    \/ represents the solidus character (U+002F).
    \b represents the backspace character (U+0008).
    \f represents the form feed character (U+000C).
    \n represents the line feed character (U+000A). 
    \r represents the carriage return character (U+000D). 
    \t represents the character tabulation character (U+0009).
```

Wierdly, the forward slash is the only character than _can_ be escaped but _need not_ be. If I'm reading the spec right.